From: Jerome Marchand <jmarchan@redhat.com>
Date: Mon, 4 May 2009 15:39:21 +0200
Subject: [block] fix request flags
Message-id: 49FEF009.1000806@redhat.com
O-Subject: [Patch RHEL5.4 1/2 v2] BZ484158: Fix request flags
Bugzilla: 484158
RH-Acked-by: Anton Arapov <aarapov@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Larry Woodman <lwoodman@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

Bugzilla:
https://bugzilla.redhat.com/show_bug.cgi?id=484158

Description:
Fix rq_flag_bits and rq_flags before I add REQ_IO_STAT flag.

Upstream status:
Upstream is not concerned.

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=1782554

Test status:
Built on all arch and successfully tested by me on i686.

Regards,
Jerome

diff --git a/block/ll_rw_blk.c b/block/ll_rw_blk.c
index f9a794f..5d1da32 100644
--- a/block/ll_rw_blk.c
+++ b/block/ll_rw_blk.c
@@ -1281,6 +1281,10 @@ static const char * const rq_flags[] = {
 	"REQ_PM_RESUME",
 	"REQ_PM_SHUTDOWN",
 	"REQ_ORDERED_COLOR",
+	"REQ_RW_SYNC",
+	"REQ_FAILFAST_DEV",
+	"REQ_FAILFAST_TRANSPORT",
+	"REQ_FAILFAST_DRIVER",
 };
 
 void blk_dump_rq_flags(struct request *rq, char *msg)
diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index f819808..fcfa093 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -236,10 +236,10 @@ enum rq_flag_bits {
 	__REQ_PM_SHUTDOWN,	/* shutdown request */
 	__REQ_ORDERED_COLOR,	/* is before or after barrier */
 	__REQ_RW_SYNC,		/* request is sync (O_DIRECT) */
-	__REQ_NR_BITS,		/* stops here */
 	__REQ_FAILFAST_DEV,	/* no driver retries of device errors */
 	__REQ_FAILFAST_TRANSPORT, /* no driver retries of transport errors */
 	__REQ_FAILFAST_DRIVER,	/* no driver retries of driver errors */
+	__REQ_NR_BITS,		/* stops here */
 };
 
 #define REQ_RW		(1 << __REQ_RW)
